plugins/scaffolder/src/components/MultistepJsonForm/MultistepJsonForm.tsx

```diff
@@ -116,13 +120,53 @@ export const MultistepJsonForm = ({
   const [activeStep, setActiveStep] = useState(0);
   const [disableButtons, setDisableButtons] = useState(false);
   const errorApi = useApi(errorApiRef);
+  const featureFlagApi = useApi(featureFlagsApiRef);
+  const featureFlagKey = 'backstage:featureFlag';
+  const filterOutProperties = (schema: JsonObject): JsonObject => {
+    const removedPropertyKeys: Array<string> = [];
+    if (schema.properties) {
+      schema.properties = Object.fromEntries(
+        Object.entries(schema.properties).filter(([key, value]) => {
+          if (value[featureFlagKey]) {
+            if (featureFlagApi.isActive(value[featureFlagKey])) {
+              return true;
+            }
+            removedPropertyKeys.push(key);
+            return false;
+          }
+          return true;
+        }),
+      );
+
+      // remove the feature flag property key from required if they are not active
+      if (Array.isArray(schema.required) && removedPropertyKeys.length > 0) {
+        for (const property of removedPropertyKeys) {
```

Comment:
could this be expressed just as `schema.required = schema.required.filter(r => !removedPropertyKeys.includes(r))`?

Reply:
Only thing about this is that `schema` is the type of `JsonValue | undefined` so I have to include the `Array.isArray(schema.required)` , but I updated it with a `.filter`.